NumberTheory/GCD.cpp

```diff
@@ -0,0 +1,33 @@
+/*
+    Euclid's GCD algorithm:
+    Basic Euclidean Algorithm is used to find GCD of two numbers say a and b. 
+    The Euclidean algorithm is based on the principle that the greatest common divisor
+    of two numbers does not change if the larger number is replaced by its difference with
+    the smaller number.
+    Below is a recursive C++ function to evaluate gcd using Euclid’s algorithm.
+    
+    Time complexity:
+    O(Log min(a, b)), where a and b are the input integers    
+    
+    Space complexity:
+    Recursive: O(log((smaller number))
```

Comment:
Line 10 : "O(Log min(a, b))"
Line 13 "O(log((smaller number))"

Keep the notation uniform, I would suggest you to keep the one on Line 10. With lowercase 'L' for log.